hw01-gradle/src/main/java/ru/otus/hw3/DIYarrayList.java

```diff
@@ -0,0 +1,231 @@
+package ru.otus.hw3;
+
+import java.util.*;
+
+public class DIYarrayList<T> implements List<T> {
+
+    private Object[] items;
+    private int size;
+
+    public DIYarrayList(int size) {
+        this.size = size;
+        items = new Object[]{};
+    }
+
+    public DIYarrayList() {
+        this(0);
+    }
+
+    @Override
+    public int size() {
+        return size;
+    }
+
+    @Override
+    public boolean isEmpty() {
+        return size == 0;
+    }
+
+    @Override
+    public boolean contains(Object o) {
+        if (o == null) return false;
+        if (size == 0) return false;
+        for (int i = 0; i < size; i++) {
+            if (o.equals(items[i])) return true;
+        }
+        return false;
+    }
+
+    @Override
+    public Iterator<T> iterator() {
+        throw new UnsupportedOperationException();
+    }
+
+    @Override
+    public Object[] toArray() {
+        Object[] buf = new Object[size];
+        if (size > 0) System.arraycopy(items, 0, buf, 0, size);
+        return buf;
+    }
+
+    @Override
+    public <T1> T1[] toArray(T1[] a) {
+        throw new UnsupportedOperationException();
+    }
+
+    @Override
+    public boolean add(T t) {
+        if (t == null) throw new IllegalArgumentException();
```

Comment:
В стандартный `ArrayList` можно добавлять null, но ваше право запретить.
В исключение лучше добавить информацию о том, что не так с аргументом, чтобы быстрее понять в чем ошибка.

```suggestion
        if (t == null) throw new IllegalArgumentException("t must be not null"");
```

В других подобных местах сами посмотрите.

Reply:
внёс исправление